cscs-checks/tools/profiling_and_debugging/gperftools_mpi_omp.py

```diff
@@ -0,0 +1,98 @@
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'daint:mc',
+                              'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.prgenv_flags = {
+            'PrgEnv-cray': ['-g', '-h nomessage=3140', '-homp', '-O2'],
+            'PrgEnv-gnu': ['-g', '-fopenmp', '-O2'],
+            'PrgEnv-intel': ['-g', '-openmp', '-O2'],
+            'PrgEnv-pgi': ['-g', '-mp', '-O2']
+        }
+        # external pprof is needed to avoid "stack trace depth >= 2**32" errors
+        self.modules = ['gperftools', 'graphviz', 'pprof']
+        self.build_system = 'Make'
+        self.iterations = 500
+        self.build_system.cppflags = [
+            '-DUSE_MPI',
+            '-D_CSCS_ITMAX=%s' % self.iterations,
+        ]
+        if lang == 'Cpp':
+            self.sourcesdir = os.path.join('src', 'C++')
+        else:
+            self.sourcesdir = os.path.join('src', lang)
+
+        if lang == 'F90':
+            self.build_system.max_concurrency = 1
+
+        self.num_tasks = 96
+        self.num_tasks_per_node = 24
+        self.num_cpus_per_task = 1
+        self.num_tasks_per_core = 2
+        self.num_iterations = self.iterations
+        self.split_file = '0.sh'
+        self.executable = self.split_file
+        self.exe = './jacobi'
+        self.rpt_file = 'gperftools.rpt'
+        self.rpt_file_txt = '%s.txt' % self.rpt_file
+        self.rpt_file_pdf = '%s.pdf' % self.rpt_file
+        self.rpt_file_doc = '%s.doc' % self.rpt_file
+        self.variables = {
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+            'ITERATIONS': str(self.num_iterations),
+            'OMP_PROC_BIND': 'true',
+            'CRAYPE_LINK_TYPE': 'dynamic',
+        }
+        self.pre_run = [
+            'echo \'#!/bin/bash\' &> %s' % self.split_file,
+            'echo \'CPUPROFILE=`hostname`.$SLURM_PROCID\' %s >> %s'
+            % (self.exe, self.split_file),
+            'chmod u+x %s' % (self.split_file),
+        ]
+        self.post_run = [
+            'pprof --text --lines %s %s &> %s'
+            % (self.exe, '*.0', self.rpt_file_txt),
+            'pprof --pdf %s %s &> %s' % (self.exe, '*.0', self.rpt_file_pdf),
+            'file %s &> %s'           % (self.rpt_file_pdf, self.rpt_file_doc)
+        ]
+        self.sanity_patterns = sn.all([
+            # check job status:
+            sn.assert_found('SUCCESS', self.stdout),
+            # check txt report:
+            sn.assert_found('MPI_Allreduce', self.rpt_file_txt),
+            # check pdf report:
+            sn.assert_found('PDF document', self.rpt_file_doc),
+        ])
+        self.perf_patterns = {
+            'hotspot1': sn.extractsingle(
+                r'^\s+\d+ms\s+(?P<flatPercentage>\d+.\d+)%.*_jacobi.\w+:\d+',
+                self.rpt_file_txt, 'flatPercentage', float)
+        }
+        self.reference = {
+            'daint:mc':  {'hotspot1': (26.0, -0.6, None, '%')},
```

Comment:
It seems that for the F90 tests the resulting percentage is consistently much higher than the Cpp ones. Maybe you should differentiate the `reference` based on the `lang`

Reply:
i would rather keep it short, there is no need to be too strict here, i only want to check that a percentage is reported.

Reply:
Then maybe you don't need to add a `reference` just put it in the sanity and check that a indeed a percentage is reported.